### PR TITLE
Fix  update proxy contracts with delete

### PR DESCRIPTION
**Pull request by alenmestrov**
Fixed queries and parsing of responses for ICP Proxy contract, introduced delete proposal action for all contracts

### CODE REVIEW

File: crates/context/config/src/client/env/proxy/query/active_proposals.rs [hunk 1]
```diff
@@ -65,7 +65,7 @@ impl Method<Icp> for ActiveProposalRequest {
     }
 
     fn decode(response: Vec<u8>) -> eyre::Result<Self::Returns> {
-        let value = Decode!(&response, Self::Returns)?;
-        Ok(value)
+        let value = Decode!(&response, u32)?;
```
Comment (frdomovic) [inline crates/context/config/src/client/env/proxy/query/active_proposals.rs:68]: @miraclx its okay for now as we are not hitting limit for u32 -> u16 conversion ?

**Review by frdomovic — APPROVED** [review]
lgtm

File: crates/context/config/src/client/env/proxy/query/active_proposals.rs (hunk 1 above)
Comment (miraclx) [reply crates/context/config/src/client/env/proxy/query/active_proposals.rs:68]: but why is it `u16` anyway, if all the contracts return `u32`?

File: crates/context/config/src/client/env/proxy/query/proposal_approvers.rs [hunk 2]
```diff
@@ -104,15 +104,17 @@ impl Method<Icp> for ProposalApproversRequest {
...
 
         // safety: `ICRepr<T>` is a transparent wrapper around `T`
         #[expect(
             clippy::transmute_undefined_repr,
             reason = "ICRepr<T> is a transparent wrapper around T"
         )]
         let identities = unsafe {
-            mem::transmute::<Vec<ICRepr<ContextIdentity>>, Vec<ContextIdentity>>(identities)
+            mem::transmute::<Vec<ICRepr<ContextIdentity>>, Vec<ContextIdentity>>(
+                identities.expect("error unwrapping identities"),
```
Comment (miraclx) [inline crates/context/config/src/client/env/proxy/query/proposal_approvers.rs]: when `None` is a valid return value, it needs to be propagated, not panicked.

```suggestion
        let Some(identities) = Decode!(&response, Option<Vec<ICRepr<ContextIdentity>>>)? else {
            return Ok(None);
        };

        // safety: `ICRepr<T>` is a transparent wrapper around `T`
        #[expect(
            clippy::transmute_undefined_repr,
            reason = "ICRepr<T> is a transparent wrapper around T"
        )]
        let identities = unsafe {
            mem::transmute::<Vec<ICRepr<ContextIdentity>>, Vec<ContextIdentity>>(identities
```

File: crates/sdk/src/env/ext.rs [hunk 3]
```diff
@@ -147,6 +153,14 @@ impl DraftProposal {
         self
     }
 
+    /// Add an action to delete a proposal.
+    #[must_use]
+    pub fn delete_proposal(mut self, proposal_id: ProposalId) -> Self {
```
Comment (miraclx) [inline crates/sdk/src/env/ext.rs]: can we just call it `delete` since the path would already be `Self::external().propose().delete(..)`

**Review by miraclx — APPROVED** [review]

**Pull request merged**